### ISSUE TITLE
Gem install no longer installing latest version

### ISSUE THREAD

**Issue by sp-ricard-valverde** [rubygems/rubygems#4521]
I'm having a problem or would like to suggest a feature.

My current problem is that when installing a gem without specifying a version(in this case, `fastlane` gem) it would install `2.54.1` instead of `2.180.1` when determining the latest version.

This issue is related to:

  - [ ] Network problems
  - [ ] Installing a library
  - [ ] Publishing a library
  - [x] The command line `gem`
  - [ ] Other

Here are my current environment details:

```
$ gem env version
3.2.15
```

Checked with gem version `3.1.2` and install works as expected.

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

**Comment by deivid-rodriguez** [comment]
Hi! The client shoud probably better information about this when falling back to older gems, but the reason is that the latest version is not installable.

In your case, I assume you're using ruby 3, which is not yet supported by the latest `fastlane`. For example, fastlane-2.180.1 depends on `google-api-client >= 0.37.0, < 0.39.0`, but all `google-api-client` versions matching that have a `required_ruby_version ~> 2.4` which does not match ruby 3.

**Comment by sp-ricard-valverde** [comment]
Many thanks for the prompt response, you're totally right.

To be honest I kind of expected to fail if the requirements did not match, but I was completely blind without any kind of warning nor debugging messages saying gem failed to match the requirements for the selected(latest) version.

**Comment by deivid-rodriguez** [comment]
Yeah, we should improve this situation by explaining why we're falling back to an older version. This is also related to https://github.com/rubygems/rubygems/issues/4338, because `gem install fastline:2.180.1` also just crashes instead of giving a proper error.
